object-manipulation-2/src/omit.js

```diff
@@ -1,15 +1,15 @@
 /* exported omit */
 // Make storage for the output
 // check each value of the array and see if it matches the properties of the source.
-// if a property matches an array value, omit the properties and store the remain propeties into storage
-// check for edge cases such as null, undefined, or empty object literals.
+// if a property does not match an array value, omit the properties and store the remaining propeties into storage
+// return the storage as output
 function omit(source, key) {
   var objectOutput = {};
   for (var i = 0; i < key.length; i++) {
     for (var prop in source) {
-      if (prop === key[i] || prop === key[i + 1] || prop === key[0]) {
-        objectOutput;
-      } else objectOutput[prop] = source[prop];
+      if (prop !== key[i] && prop !== key[i + 1] && prop !== key[0]) {
```

Comment:
I don't understand your logic for this condition. Can you explain it to me? How does this check if the current key is not present in the object? Does this function still pass all tests?

Reply:
my logic was that in order to check  the array and properties together we need both loops, however because if i nest them it does not do one iteration of properties but rather checks one property for both key values which kept giving me more values in my output than intended so to pinpoint just the one true value that was not in the array i needed a condition that made sure that the  property in the loop was not in any of the array values.